android/src/com/mapswithme/maps/widget/placepage/PlaceOpeningHoursAdapter.java

```diff
@@ -107,7 +146,26 @@ public void onBindViewHolder(@NonNull ViewHolder holder, int position)
   @Override
   public int getItemCount()
   {
-    return (mTimetables != null ? mTimetables.length : 0) + (closedDays == null ? 0 : 1);
+    return (mWeekSchedule != null ? mWeekSchedule.size() : 0);
+  }
+
+  public static class WeekScheduleData
+  {
+    public final int startWeekDay;
+    public final int endWeekDay;
+    public final boolean isClosed;
+    public final Timetable timetable;
+
+    public WeekScheduleData(int startWeekDay, int endWeekDay, boolean isClosed, Timetable timetable)
```

Comment:
Looks like it duplicates the Timetable structure. Is it really necessary? No Timetable for a day means that it's closed, no?

Reply:
Single timetable could be splitted on UI.
E.g. `"Mo,Sa,Su 10:00-18:00"`. This string is parsed as a single Timetable instance. But on UI it would be 2 lines:

```
Mo      10:00-18:00
Tu-Fr   closed
Sa-Su   10:00-18:00
```
Or
```
Su-Mo   10:00-18-00
Tu-Fr   closed
Sa      10:00-18:00
```

To reflect this structure I created `WeekScheduleData` class. For each row I store dates range and link to a Timetable.